src/machine_learning/k_means.jl

```diff
@@ -0,0 +1,126 @@
+"""
+   A K-Means algorithm for data clustering
+
+# Brief:
+    The algorithm takes n-dimensional points represented as a vector where each element of that vector
+    corresponds to one dimension of the point. In the beginning, the algorithm select k points.
+    Every of the k points is used as a centre of the cluster. In each iteration, every of these k points
+    is connected to the nearest points and moved, until it stops changing its position through iterations.
+
+# For more informations:
+    https://en.wikipedia.org/wiki/K-means_clustering
+
+# Contributed by:- [Nikola Mircic](https://github.com/Nikola-Mircic)
+"""
+
+module KMeans
+# A function that takes two points and returns the distance 
+# between them
+function distance(a::Vector{T}, b::Vector{T}) where T<:Number
+    if length(a) != length(b)
+        return 0.0;
+    end
+    sum = 0.0
```

Comment:
try not to use `sum` as a variable name since it is already in Julia Base. It may cause confusion.